File: FSharpTranslate/Logger.fs
module Logger

open System
open System.IO

let private lockObj = obj()
let private logFile = Path.Combine(AppContext.BaseDirectory, "fsharptranslate.log")

let private write level message =
    try
        lock lockObj (fun () ->
            let logEntry = sprintf "[%s] %s: %s%s" (DateTime.UtcNow.ToString("O")) level message Environment.NewLine
            File.AppendAllText(logFile, logEntry)
        )
    with
    | _ -> () // Best effort logging; don't crash app

let info message = write "INFO" message
let error message = write "ERROR" message
let debug message = write "DEBUG" message

File: FreeTranslateWin/Logger.cs
using System;
using System.IO;

namespace TranslationFiesta
{
    public static class Logger
    {
        private static readonly object _lock = new object();
        private static readonly string LogFile = Path.Combine(AppContext.BaseDirectory, "translationfiesta.log");

        public static void Info(string msg) => Write("INFO", msg);
        public static void Error(string msg) => Write("ERROR", msg);
        public static void Debug(string msg) => Write("DEBUG", msg);

        private static void Write(string level, string msg)
        {
            try
            {
                lock (_lock)
                {
                    File.AppendAllText(LogFile, $"[{DateTime.UtcNow:O}] {level}: {msg}{Environment.NewLine}");
                }
            }
            catch
            {
                // Best effort logging; don't crash app
            }
        }
    }
}
